src/metabase/driver/ddl/postgres.clj

```diff
@@ -0,0 +1,114 @@
+(ns metabase.driver.ddl.postgres
+  (:require [clojure.java.jdbc :as jdbc]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [metabase.driver.ddl.interface :as ddl.i]
+            [metabase.driver.sql-jdbc.connection :as sql-jdbc.conn]
+            [metabase.driver.sql.util :as sql.u]
+            [metabase.public-settings :as public-settings]
+            [metabase.query-processor :as qp]
+            [metabase.util.i18n :refer [trs]]))
+
+(defn- quote-fn [driver]
+  (fn quote [ident entity]
+    (sql.u/quote-name driver ident (ddl.i/format-name driver entity))))
+
+(defn- create-schema-sql
+  "SQL string to create a schema suitable for postgres"
+  [{driver :engine :as database}]
+  (let [q (quote-fn driver)]
+    (format "create schema %s"
+            (q :table (ddl.i/schema-name database (public-settings/site-uuid))))))
```

Comment:
A lot of this stuff can be done in HoneySQL especially HoneySQL 2 these days. Not sure if that's a lot easier or not but it's definitely a little nicer to work with than manually trying to create the SQL for DDL stuff